### ISSUE TITLE
Using source in another database results in an error

### ISSUE THREAD

**Issue by veeti** [ClickHouse/dbt-clickhouse#138]
Hi,

I'm trying to define [DBT source tables](https://docs.getdbt.com/docs/build/sources) that are located in another database from my project:

```yml
# profiles.yml
profile:
  outputs:
    xyz:
      schema: some_database

# models/something.yml
sources:
  - name: xyz
    database: some_other_database
    tables:
      - name: abc
      - name: def
```

This results in the error "Cannot set database some_other_database in Clickhouse!" due to [this check](https://github.com/ClickHouse/dbt-clickhouse/blob/36045655ca743d7ad482ae852fd277aa3ac1ad1c/dbt/adapters/clickhouse/relation.py#L30).

Is it possible to add support for sources in other databases or is there something I'm doing wrong?

**Comment by genzgd** [comment]
Try setting `schema` at the top level of your sources and database in the table element - this works for me?  

```yml
sources:
  - name: taxis_source
    schema: taxis
    description: This references the main taxi table as copied from S3
    tables:
      - database: taxis
        name: trips
```

I agree that database should actually work at that level but this should let you work around it.

**Comment by BentsiLeviav** [comment]
Hi @veeti,

It appears that this issue has been resolved as part of the development process for other features. Currently, all methods responsible for creating a relationship instance pass an empty string as the database (e.g., [create_from_source](https://github.com/ClickHouse/dbt-clickhouse/blame/main/dbt/adapters/clickhouse/relation.py#L92) and [create_from_node](https://github.com/ClickHouse/dbt-clickhouse/blame/main/dbt/adapters/clickhouse/relation.py#L118)). Consequently, this check will no longer trigger an exception.

At the time this issue was initially opened, the most recent version did not incorporate this implementation; instead, it relied on the implementation provided by `BaseRelation`.

In any case, I ran your scenario with the latest version, and it worked smoothly.
